src/Application/Matching/Adapter.php

```diff
@@ -56,14 +52,14 @@ public function __construct(
      */
     public function addAmount(CampaignFunding $funding, string $amount): string
     {
-        $fundBalance = $this->addAmountWithoutSavingFundingsToDB($funding, $amount);
-        $this->saveFundingsToDatabase();
-
-        return $fundBalance;
+        return $this->addAmountWithoutSavingFundingsToDB($funding, $amount);
```

Comment:
IMHO we'd be better to rename this, or better inline it into `Matching\Adapter::addAmount`. 

Deferred implicit is the doctrine default, I don't see a reason to mention it just here. There's nothing special about this function not calling 'flush', we would put `WithoutSavingToDB` all over our codebase if we wanted to mark all functions that don't flush.

Reply:
Particularly since `addAmount` also doesn't do its own flush.

Reply:
Done, more on overlapping thread at https://github.com/thebiggive/matchbot/pull/1144#pullrequestreview-2514010257